modules/build/src/main/scala/scala/build/options/ConfigMonoid.scala

```diff
@@ -81,9 +81,13 @@ object ConfigMonoid {
       main ++ defaults
   }
 
+  implicit def map[K, V]: ConfigMonoid[Map[K, V]] = instance(Map.empty[K, V]) {
+    (main, defaults) =>
+      main ++ defaults
+  }
```

Comment:
This discards the values in `main` whose keys are also in `defaults`, which is not what we want. Something like this should do:
```scala
implicit def map[K, V](implicit valueMonoid: ConfigMonoid[V]): ConfigMonoid[Map[K, V]] =
  instance(Map.empty[K, V]) { (main, defaults) =>
    (main.keySet ++ defaults.keySet)
      .iterator
      .map { k =>
        val v = valueMonoid.orElse(main.getOrElse(k, valueMonoid.zero), defaults.getOrElse(k, valueMonoid.zero))
        k -> v
      }
      .toMap
  }
```

Reply:
Done